extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SelectionRenderer.java

```diff
@@ -72,11 +75,18 @@ public boolean canRender(Widget w) {
      *
      * @return String representing JSON object
      */
-    private String getMappingsJSON(Selection w) {
+    private String getMappingsJSON(Selection w, Item item) {
         JsonObject resultObject = new JsonObject();
-        for (Mapping mapping : w.getMappings()) {
-            resultObject.addProperty(mapping.getCmd(), mapping.getLabel());
+        if (w.getMappings().size() == 0 && item != null && item.getStateDescription() != null) {
+            for (StateOption option : item.getStateDescription().getOptions()) {
```

Comment:
It looks like we iterate over all state options or mappings twice. One time in this method (getMappingsJSON) the second time in the renderWidget method. Do you think it is possible to speed up the process by eliminating one loop? E.g. is the getMappingsJSON method imho not really necessary.

Reply:
Yes, that is possible of course.